src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -1538,6 +1527,100 @@ bool RegisterAppInterfaceRequest::GetDataOnSessionKey(
   return true;
 }
 
+mobile_apis::Result::eType
+RegisterAppInterfaceRequest::ProcessingAppHMITypesInMessage(
+    const smart_objects::SmartObject& message) {
+  const bool app_hmi_types_exist =
+      message[strings::msg_params].keyExists(strings::app_hmi_type);
+
+  if (!app_hmi_types_exist) {
+    return mobile_apis::Result::SUCCESS;
+  }
+
+  auto app_types_in_message =
+      message[strings::msg_params][strings::app_hmi_type];
+
+  auto app_types_in_message_array = *(app_types_in_message.asArray());
+
+  const bool check_result = (app_types_in_message_array.end() !=
+                             std::find(app_types_in_message_array.begin(),
+                                       app_types_in_message_array.end(),
+                                       mobile_apis::AppHMIType::WEB_VIEW));
+  return check_result ? mobile_apis::Result::DISALLOWED
+                      : mobile_apis::Result::SUCCESS;
+}
+
+// mobile_apis::Result::eType
+// RegisterAppInterfaceRequest::ProcessingAppHMITypesInMessage(
+//    const smart_objects::SmartObject& message) {
+//  const bool app_hmi_types_exist =
+//      message[strings::msg_params].keyExists(strings::app_hmi_type);
+//  if (!app_hmi_types_exist) {
+//    return mobile_apis::Result::SUCCESS;
+//  }
+//  const std::string mobile_app_id =
+//      application_manager_.GetCorrectMobileIDFromMessage(message_);
+
+//  const auto& app_types_in_message =
+//      message[strings::msg_params][strings::app_hmi_type];
+//  const bool check_result = GetPolicyHandler().CheckHMIType(
+//      mobile_app_id, mobile_apis::AppHMIType::WEB_VIEW,
+//      &app_types_in_message);
+
+//  return check_result ? mobile_apis::Result::SUCCESS
+//                      : mobile_apis::Result::DISALLOWED;
+//}
+
+mobile_apis::Result::eType
+RegisterAppInterfaceRequest::ProcessingAppHMITypesPolicies(
+    smart_objects::SmartObject& message,
+    policy::StringArray& app_hmi_types_in_policy) {
+  auto result = mobile_apis::Result::SUCCESS;
+  const bool app_hmi_types_exist =
+      message[strings::msg_params].keyExists(strings::app_hmi_type);
+
+  if (app_hmi_types_exist) {
+    // If AppHmiTypes are partially same, the system should allow those listed
+    // in the policy table and send warning info on missed values
+    auto app_types_in_message =
+        *(message[strings::msg_params][strings::app_hmi_type].asArray());
+
+    const std::string mobile_app_id =
+        application_manager_.GetCorrectMobileIDFromMessage(message_);
+    std::string log;
+    CheckMissedTypes checker(mobile_app_id, GetPolicyHandler(), log);
+    std::for_each(
+        app_types_in_message.begin(), app_types_in_message.end(), checker);
+    if (!log.empty()) {
+      result = ProcessingAppHMITypesInMessage(message);
```

Comment:
result = log.find("WEB_VIEW") != std::string::npos ? mobile_apis::Result::DISALLOWED : mobile_apis::Result::WARNINGS;

Reply:
This would remove the need for ProcessingAppHMITypesInMessage and re-parsing the message contents. The log string contains the necessary information needed for finding out if WEB_VIEW app hmi type was used without policy permission.